mgmt/rest/v2/task.go

```diff
@@ -49,8 +109,10 @@ type Task struct {
 	MissCount          int               `json:"miss_count,omitempty"`
 	FailedCount        int               `json:"failed_count,omitempty"`
 	LastFailureMessage string            `json:"last_failure_message,omitempty"`
-	State              string            `json:"task_state"`
-	Href               string            `json:"href"`
+	State              string            `json:"state,omitempty"`
+	Href               string            `json:"href,omitempty"`
+	Start              bool              `json:"start,omitempty"`
+	MaxFailures        int               `json:"max-failures,omitempty"`
```

Comment:
Consolidated `Task` data model from Snap CLI and Rest API so that they reference the same `Task` data model in Swagger spec. The benefit is that no business logic in CLIs/clients and Snap server defines the client behaviors universally.